pwiz/analysis/spectrum_processing/SpectrumList_IonMobility.cpp

```diff
@@ -98,6 +99,14 @@ SpectrumList_IonMobility::SpectrumList_IonMobility(const msdata::SpectrumListPtr
                 units_ = IonMobilityUnits::compensation_V;
             else if (!scan.userParam("drift time").empty()) // Oldest known mzML drift time style
                 units_ = IonMobilityUnits::drift_time_msec;
+            for (int b = 0; b < spectrum->binaryDataArrayPtrs.size(); b++)
+            {
+                if (spectrum->binaryDataArrayPtrs[b]->hasCVParam(CVID::MS_raw_ion_mobility_array))

```

Comment:
I did this a little more robustly in the MBI PR: https://github.com/ProteoWizard/pwiz/pull/1484/files#diff-03d7cf0af6b548cf602001d8c4932f32ab89c6be17131f85efdaa32738f1dc35

I think you can copy that bit here. You'll get the unit as well, although another else branch is needed to support the new SONAR scheme (not using ion mobility cvParam). For some reason I didn't do the change to hasCombinedIonMobility, was that one of the reasons it was acting up in Skyline?

Reply:
I was trying to evaluate the effect of SONAR filtering by comparing it to mzML from a current msconvert, which produces a drift time representation. Loading that into Skyline with no IMS filtering enabled was the plan, but Skyline wouldn't read it at all because the pwiz layer was only recognizing 3-array data in native formats.

I'm not yet worrying about proper SONAR representation in mzML, initial focus is on raw file handling so I can get something to Waters. But yeah, that will need to be addressed.

Reply:
Huh, in the MBI PR code is a "MS_raw_ion_mobility_array" data array really a marker for drift time, specifically?

Reply:
Yes, raw ion mobility array is only drift time.